### ISSUE TITLE
Fix a11y issues in components

### ISSUE THREAD

**Issue by jedgar1mx** [CityOfDetroit/COD-Design-System#374]
**Describe the bug**
Some components are triggering a11y errors when used on our sites.

**To Reproduce**

1. Go to site
2. Run a11y scan
3. See error

**Expected behavior**
A11y scan should be clear

**Screenshots**

<img width="1040" height="320" alt="Image" src="https://github.com/user-attachments/assets/49515584-cae4-40e0-a7b7-acd713f82872" />
